internal/events/handlers.go

```diff
@@ -31,7 +31,12 @@ func (s *Service) AddEvent(w http.ResponseWriter, r *http.Request) {
 	defer r.Body.Close()
 
 	// check whether event already exists
-	name := s.db.TransferName(ev.Contract)
+	name, flag := s.db.TransferName(ev.Contract)
+	if flag {
+		w.WriteHeader(http.StatusForbidden)
```

Comment:
for a request response complaining about badly formatted parameters, I would return a 400 instead. Bad request makes more sense to me.